scripts/setup_test.py

```diff
@@ -107,9 +107,9 @@ def mock_get(unused_var):
         self.get_swap = self.swap(os.environ, 'get', mock_get)
         self.cd_swap = self.swap(common, 'CD', MockCD)
         version_info = collections.namedtuple(
-            'version_info', ['major', 'minor'])
-        self.version_info_py37_swap = self.swap(
-            sys, 'version_info', version_info(major=3, minor=7)
+            'version_info', ['major', 'minor', 'micro'])
+        self.version_info_py38_swap = self.swap(
+            sys, 'version_info', version_info(major=3, minor=8, micro=12)
```

Comment:
I think the canonical name for the last part of the version number is "patch" instead of "micro" (see https://semver.org/)

Reply:
![2022 06 23-14 39 33 screenshot](https://user-images.githubusercontent.com/63491234/175262708-f1b1e419-f656-4bad-9b0e-3dcc1b202a8e.png)
![2022 06 23-14 43 08 screenshot](https://user-images.githubusercontent.com/63491234/175263512-cb447aef-4309-45ef-9a2a-ad9c11fc9787.png)
![2022 06 23-14 54 13 screenshot](https://user-images.githubusercontent.com/63491234/175265840-b937d163-6a31-45c6-a260-2b34dafbce12.png)

 
@U8NWXD  So i went through it again to double check. It turns out the output of sys.version_info is a named tuple with major, minor, micro, release level, serial version/number. But what you're stating is also right. As you can see in the 2nd image the `micro` data descriptor is actually the `patch release number`. However since the named value of descriptors defined is `micro` and not `patch` , i think we will have to use it . In the third image you can see the command produces an error if i try to fetch the `patch release number` using patch named descriptor.